### DISCUSSION TITLE
Email will not send when form is filled out on IOS device

### DISCUSSION THREAD

**Discussion by paul-fitzpatrick** [emailjs-com/help#17] [Q&A]
I have a django app, which send an email when a form is submitted. the email send as intended when I fill out and submit the form on an android device or desktop, but does not send the email when the form is submitted on an ipad.
I've pasted in my JS which sends the mail, is there any tweaks that need to be made to suit IOS?
I'm a relatively dev so apologies if I'm asking something which has an obvious answer!

<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/@emailjs/browser@3/dist/email.min.js"></script>
    <script type="text/javascript">
        (function() {
            // https://dashboard.emailjs.com/admin/account
            emailjs.init('wisaRMYSfqHIVa0mN');
        })();
    </script>

    <script type="text/javascript">
        window.onload = function() {
            document.getElementById('contact-form').addEventListener('submit', function(event) {
                // event.preventDefault();
                // generate a five digit number for the contact_number variable
                this.contact_number.value = Math.random() * 100000 | 0;
                // these IDs from the previous steps
                emailjs.sendForm('contact_service', 'contact_form', this)
                    .then(function() {
                        console.log('SUCCESS!');
                    }, function(error) {
                        console.log('FAILED...', error);
                    });
            });
        }
    </script>

**Comment by xr0master** [comment]
Well, why did you comment out part of the code?

**Reply by paul-fitzpatrick** [reply]
So until I commented out event.preventDefault(); the email would send (on everthing but IOS) but the form wouldn't go through to the django admin site. once I remove that line, all the form info goes through to the admin as it should.

**Reply by xr0master** [reply]
If you need to pass this form data to your server code, in that case I recommend moving the EmailJS code to your server. You can use our REST API to send the request from python.

**Reply by paul-fitzpatrick** [reply]
ok so i've added from the rest api page :

And I added the template id, service id and user id environment variables to my project on render.com

It doesn't send at all now.
Did i miss something? or maybe I haven't actually it right?

<script>
        $('#contact-form').on('submit', function(event) {
    //event.preventDefault(); // prevent reload
    
    var formData = new FormData(this);
    formData.append('service_id', 'default_service');
    formData.append('template_id', 'contact-form');
    formData.append('user_id', 'my actual public key');
 
    $.ajax('https://api.emailjs.com/api/v1.0/email/send-form', {
        type: 'POST',
        data: formData,
        contentType: false, // auto-detection
        processData: false // no need to parse formData to string
    }).done(function() {
        alert('Your mail is sent!');
    }).fail(function(error) {
        alert('Oops... ' + JSON.stringify(error));
    });
});
    </script>

**Reply by paul-fitzpatrick** [reply]
apologies, I didn't realize that text would display so badly once I hit enter! I used the code for https://www.emailjs.com/docs/rest-api/send-form/  but obviously change my template, service and user id's

**Reply by xr0master** [reply]
What I meant was to make a server-to-server call. Send requests to EmailJS from your server, after or in parallel, as you have processed the form data.
